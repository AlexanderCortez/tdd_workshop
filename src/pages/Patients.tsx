import React, { useCallback, useEffect, useState } from 'react';
import PatientsList from 'components/patientsList/PatientsList';
import { IPatient } from 'types/interfaces/patient.interface';
import { getPatients } from 'actions/patient.action';

const Patients = (): React.ReactElement => {
  const [patients, setPatients] = useState<IPatient[]>([]);

  const loadPatients = useCallback(async (): Promise<void> => {
    const fetchedPatients = await getPatients();
    setPatients(fetchedPatients);
  }, []);

  useEffect(() => {
    loadPatients();
  }, [loadPatients]);

  return (
    <div>
      <h2>Patients</h2>
      <PatientsList patients={patients} />
    </div>
  );
};

export default Patients;
