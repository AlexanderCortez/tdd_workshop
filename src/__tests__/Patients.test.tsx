import React from 'react';
import { shallow } from 'enzyme';
import Patients from 'pages/Patients';
import * as patientActions from 'actions/patient.action';
import factories from '__tests__/factories';
import { sleep } from './utils';

describe('Patients Test', (): void => {
  it('should render all components', () => {
    const wrapper = shallow(<Patients />);
    expect(wrapper.find('h2').text()).toEqual('Patients');
    expect(wrapper.exists('PatientsList')).toBeTruthy();
  });

  it('should load some patients from api', async () => {
    const patients = factories.patient.buildList(8);
    jest.spyOn(patientActions, 'getPatients').mockResolvedValue(patients);
    jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());
    const wrapper = shallow(<Patients />);
    await sleep();
    const patientListComponent = wrapper.find('PatientsList');
    expect(patientListComponent.props()).toHaveProperty('patients', patients);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<Patients />);
    expect(wrapper).toMatchSnapshot();
  });
});
