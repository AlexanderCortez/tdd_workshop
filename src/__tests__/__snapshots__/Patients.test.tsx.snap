// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patients Test should match snapshot 1`] = `
<div>
  <h2>
    Patients
  </h2>
  <PatientsList
    patients={Array []}
  />
</div>
`;
